runner/views/run_api_view.py

```diff
@@ -652,6 +653,82 @@ def post(self, request):
+
+    serializer_class = ArgosPairingSerializer
+
+    def post(self, request):
+        igo_request_id = request.data.get("igo_request_id", None)
+        argos_slug = request.data.get("argos_slug", None)
+        if igo_request_id and argos_slug:
+            operator_model = Operator.objects.get(slug=argos_slug)
+            operator = OperatorFactory.get_by_model(operator_model, request_id=igo_request_id)
+            # construct_argos_jobs() is sloppily separate from the Operator module
+            from runner.operator.argos_operator.v2_0_0.construct_argos_pair import construct_argos_jobs
+
+            files, cnt_tumors = operator.get_files(operator.request_id)
+            data = operator.build_data_list(files)
+            samples = operator.get_samples_from_data(data)
+            argos_inputs, error_samples = construct_argos_jobs(samples)
+            sample_mapping, filepaths = operator.get_mapping_from_argos_inputs(argos_inputs)
+            if sample_mapping:
+                body = {"details": sample_mapping}
+            else:
+                message = "%s: No samples found." % igo_request_id
+                body = {"details": message}
+        return Response(body, status=status.HTTP_202_ACCEPTED)
+
+
+class ArgosDataClinicalViewSet(GenericAPIView):
+    logger = logging.getLogger(__name__)
+
+    serializer_class = ArgosPairingSerializer
+
+    def post(self, request):
+        igo_request_id = request.data.get("igo_request_id", None)
+        argos_slug = request.data.get("argos_slug", None)
+        if igo_request_id and argos_slug:
+            operator_model = Operator.objects.get(slug=argos_slug)
+            operator = OperatorFactory.get_by_model(operator_model, request_id=igo_request_id)
+            # construct_argos_jobs() and compile_pairs() are sloppily separate from the Operator module
+            from runner.operator.argos_operator.v2_0_0.construct_argos_pair import construct_argos_jobs
+            from runner.operator.argos_operator.v2_0_0.bin.pair_request import compile_pairs
+
+            files, cnt_tumors = operator.get_files(operator.request_id)
+            data = operator.build_data_list(files)
+            samples = operator.get_samples_from_data(data)
+            dmp_samples = list()
+            for sample in samples:
+                sample_type = sample["tumor_type"]
+                this_sample, is_dmp_sample = operator.get_regular_sample(sample, sample_type)
+                if is_dmp_sample:
+                    dmp_samples.append(this_sample)
+            argos_inputs, error_samples = construct_argos_jobs(samples)
+            sample_mapping, filepaths = operator.get_mapping_from_argos_inputs(argos_inputs)
+            dmp_samples = operator.get_dmp_samples_from_argos_inputs(argos_inputs)
+            pipeline = operator.get_pipeline_id()
+            try:
+                pipeline_obj = Pipeline.objects.get(id=pipeline)
+            except Pipeline.DoesNotExist:
+                pass
```

Comment:
return Response with error for example 404. Otherwise it will break https://github.com/mskcc/beagle/pull/1343/files#diff-ae1940d4baf50cb6d3ef8747783319716e5256c420bdd3f81be25bf51428f6dbR717 with 500

Reply:
Added